### ISSUE TITLE
UUID duplicate

### ISSUE THREAD

**Issue by ristridin01** [hp-net/homebridge-nibe#7]
There seems to be a duplicate entry, another homebridge plugin seems to use the same UUID.
Log:
[homebridge-nibe] Could not restore cached accessory 'status': Cannot add a bridged Accessory with the same UUID as another bridged Accessory: 07fd68b2-5387-4138-9d6c-d7977d40437d

**Comment by hpruszyn** [comment]
Similar issue https://github.com/homebridge/homebridge/issues/1685. However I'm still looking at it.

**Comment by hpruszyn** [comment]
I've added logs when accessory is added from cached or new accessory is discovered. @ristridin01 please update plugin and let me know if bug is still there and if yes then please attach logs from homebridge (filter only those for nibe).

**Comment by ristridin01** [comment]
not sure if that has something to do with this issue but I updated the plugin and received that error:

[4/6/2022, 7:19:20 AM] [Nibe] error from serviceinfo/categories?parameters=true&systemUnitId=0
[4/6/2022, 7:19:20 AM] [Nibe] Error: Error: getaddrinfo EAI_AGAIN api.nibeuplink.com
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
  errno: -3001,
  code: 'EAI_AGAIN',
  syscall: 'getaddrinfo',
  hostname: 'api.nibeuplink.com',
  config: {
    transitional: {
      silentJSONParsing: true,
      forcedJSONParsing: true,
      clarifyTimeoutError: false
    },
    adapter: [Function: httpAdapter],
    transformRequest: [ [Function: transformRequest] ],
    transformResponse: [ [Function: transformResponse] ],
    timeout: 45000,
    xsrfCookieName: 'XSRF-TOKEN',
    xsrfHeaderName: 'X-XSRF-TOKEN',
    maxContentLength: -1,
    maxBodyLength: -1,
    validateStatus: [Function: validateStatus],
    headers: {
      Accept: 'application/json, text/plain, */*',
      'user-agent': 'fetcher.nibeuplink',
      Authorization: 'Bearer AAEAALJXJhTEASni6YoUovkVqm54kijui12XFmEhGCAVtRhoLnRJXZMFrhawU2tlQmI-ZJOt8qltrmm_s1a24Y6RlLRPwgUlYsp8o0M3mZGnaI3tZuM6rQFmlPzF5odU-BgReZLauBHz2VVBH72Ja99-Q1XytTu0jWYlw_FKKvaiya5HlJM4kmoxtmC-am_MKaLnshz3SkRNjEGTPh2iDl-gGFK0MjEczecnF6QJPOZzJha9I06IraHXIbNsZIGy7bq0ZWWOp4Yyp0mtDSIvbIoYk2de964PkP9YXAsxMRErg2ktI2CD-0nratLqWR7fKwhIpT41Gegdf6A7cnI3uegLse20AQAAAAEAAEXhHqrOfBCGOoNVekKumciMbCFg3n8-33jvgAqVtheg-UslOxLWKgLqra_KyJanFewDpexl0TixB7k_gsEgjxCi2fhRExMGgfJiIuA4N0P-cSUCILcGmYxtxo3WP5Glj-KXqhccAX3zQvZ94ZADWrXlCYvsmcnoXVvWJCJWXa-XXdmp_PwTZ3aSq2ybKDd9zb4iTvAVpyXjfMvSVb_HOt6qbyOiJVVAmhKaEiGFsbIK4Sv0RO-GUM9bT_rKe8llS1ktXfUsWgBUJCME3AHPja0rlGMTEgZcUQx1xy0cpJ2SmCdT0E_Eqc0qCN1HFcFc3RrGl-pJvhKZL35gC2jmhMye4X4U8KiQVc4nXegPVp7pjv0PSYQWC1Zw2FyzwP6VMUDZhPbwArJyeJhc9zhrQZeDtDj-1O_FfVOfZX4qMIzDHu3BZwkv1gyH2ImwSKdI1k90rC_qhb3kyy03_gzgwSFQkuATI0UqkImPMPT1TCDIY80BIDBiJbQJ6NFrwtUvXbP8pAFqzIgkCoTtQL45wmfhPM9vo32K2q-7R7_4h_esxjNBacihF8Y7hrjhB-rx7A',
      'Accept-Language': 'en'
    },
    baseURL: 'https://api.nibeuplink.com',
    method: 'get',
    url: '/api/v1/systems/48093/serviceinfo/categories?parameters=true&systemUnitId=0',
    data: undefined
  },
  request: <ref *1> Writable {
    _writableState: WritableState {
      objectMode: false,
      highWaterMark: 16384,
      finalCalled: false,
      needDrain: false,
      ending: false,
      ended: false,
      finished: false,
      destroyed: false,
      decodeStrings: true,
      defaultEncoding: 'utf8',
      length: 0,
      writing: false,
      corked: 0,
      sync: true,
      bufferProcessing: false,
      onwrite: [Function: bound onwrite],
      writecb: null,
      writelen: 0,
      afterWriteTickInfo: null,
      buffered: [],
      bufferedIndex: 0,
      allBuffers: true,
      allNoop: true,
      pendingcb: 0,
      constructed: true,
      prefinished: false,
      errorEmitted: false,
      emitClose: true,
      autoDestroy: true,
      errored: null,
      closed: false,
      closeEmitted: false,
      [Symbol(kOnFinished)]: []
    },
    _events: [Object: null prototype] {
      response: [Function: handleResponse],
      error: [Function: handleRequestError],
      socket: [Array]
    },
    _eventsCount: 3,
    _maxListeners: undefined,
    _options: {
      maxRedirects: 21,
      maxBodyLength: 10485760,
      protocol: 'https:',
      path: '/api/v1/systems/48093/serviceinfo/categories?parameters=true&systemUnitId=0',
      method: 'GET',
      headers: [Object],
      agent: undefined,
      agents: [Object],
      auth: undefined,
      hostname: 'api.nibeuplink.com',
      port: null,
      nativeProtocols: [Object],
      pathname: '/api/v1/systems/48093/serviceinfo/categories',
      search: '?parameters=true&systemUnitId=0'
    },
    _ended: true,
    _ending: true,
    _redirectCount: 0,
    _redirects: [],
    _requestBodyLength: 0,
    _requestBodyBuffers: [],
    _onNativeResponse: [Function (anonymous)],
    _currentRequest: ClientRequest {
      _events: [Object: null prototype],
      _eventsCount: 7,
      _maxListeners: undefined,
      outputData: [],
      outputSize: 0,
      writable: true,
      destroyed: false,
      _last: true,
      chunkedEncoding: false,
      shouldKeepAlive: false,
      maxRequestsOnConnectionReached: false,
      _defaultKeepAlive: true,
      useChunkedEncodingByDefault: false,
      sendDate: false,
      _removedConnection: false,
      _removedContLen: false,
      _removedTE: false,
      _contentLength: 0,
      _hasBody: true,
      _trailer: '',
      finished: true,
      _headerSent: true,
      _closed: false,
      socket: [TLSSocket],
      _header: 'GET /api/v1/systems/48093/serviceinfo/categories?parameters=true&systemUnitId=0 HTTP/1.1\r\n' +
        'Accept: application/json, text/plain, */*\r\n' +
        'user-agent: fetcher.nibeuplink\r\n' +
        'Authorization: Bearer AAEAALJXJhTEASni6YoUovkVqm54kijui12XFmEhGCAVtRhoLnRJXZMFrhawU2tlQmI-ZJOt8qltrmm_s1a24Y6RlLRPwgUlYsp8o0M3mZGnaI3tZuM6rQFmlPzF5odU-BgReZLauBHz2VVBH72Ja99-Q1XytTu0jWYlw_FKKvaiya5HlJM4kmoxtmC-am_MKaLnshz3SkRNjEGTPh2iDl-gGFK0MjEczecnF6QJPOZzJha9I06IraHXIbNsZIGy7bq0ZWWOp4Yyp0mtDSIvbIoYk2de964PkP9YXAsxMRErg2ktI2CD-0nratLqWR7fKwhIpT41Gegdf6A7cnI3uegLse20AQAAAAEAAEXhHqrOfBCGOoNVekKumciMbCFg3n8-33jvgAqVtheg-UslOxLWKgLqra_KyJanFewDpexl0TixB7k_gsEgjxCi2fhRExMGgfJiIuA4N0P-cSUCILcGmYxtxo3WP5Glj-KXqhccAX3zQvZ94ZADWrXlCYvsmcnoXVvWJCJWXa-XXdmp_PwTZ3aSq2ybKDd9zb4iTvAVpyXjfMvSVb_HOt6qbyOiJVVAmhKaEiGFsbIK4Sv0RO-GUM9bT_rKe8llS1ktXfUsWgBUJCME3AHPja0rlGMTEgZcUQx1xy0cpJ2SmCdT0E_Eqc0qCN1HFcFc3RrGl-pJvhKZL35gC2jmhMye4X4U8KiQVc4nXegPVp7pjv0PSYQWC1Zw2FyzwP6VMUDZhPbwArJyeJhc9zhrQZeDtDj-1O_FfVOfZX4qMIzDHu3BZwkv1gyH2ImwSKdI1k90rC_qhb3kyy03_gzgwSFQkuATI0UqkImPMPT1TCDIY80BIDBiJbQJ6NFrwtUvXbP8pAFqzIgkCoTtQL45wmfhPM9vo32K2q-7R7_4h_esxjNBacihF8Y7hrjhB-rx7A\r\n' +
        'Accept-Language: en\r\n' +
        'Host: api.nibeuplink.com\r\n' +
        'Connection: close\r\n' +
        '\r\n',
      _keepAliveTimeout: 0,
      _onPendingData: [Function: nop],
      agent: [Agent],
      socketPath: undefined,
      method: 'GET',
      maxHeaderSize: undefined,
      insecureHTTPParser: undefined,
      path: '/api/v1/systems/48093/serviceinfo/categories?parameters=true&systemUnitId=0',
      _ended: false,
      res: null,
      aborted: false,
      timeoutCb: null,
      upgradeOrConnect: false,
      parser: null,
      maxHeadersCount: null,
      reusedSocket: false,
      host: 'api.nibeuplink.com',
      protocol: 'https:',
      _redirectable: [Circular *1],
      [Symbol(kCapture)]: false,
      [Symbol(kNeedDrain)]: false,
      [Symbol(corked)]: 0,
      [Symbol(kOutHeaders)]: [Object: null prototype]
    },
    _currentUrl: 'https://api.nibeuplink.com/api/v1/systems/48093/serviceinfo/categories?parameters=true&systemUnitId=0',
    _timeout: null,
    [Symbol(kCapture)]: false
  },
  response: undefined,
  isAxiosError: true,
  toJSON: [Function: toJSON]
}
[4/6/2022, 7:19:20 AM] [Nibe] error from serviceinfo/categories?parameters=true&systemUnitId=1

**Comment by hpruszyn** [comment]
This error happens when AuthCode needs to be regenerate. This code has a time to live and when it is close to an end it will refresh automatically. However if you turn off plugin, have power failure, etc that will cause problem because code can't be refreshed and you need to generate new one. It will also occur when you remove homebridge storage e.g.: you run homebridge on docker without mount from host, then with every restart storage will be lost.

In normal use it should not happen. In my "production" environment it didn't happen and plugin is running for a few days now. In my "dev" environment on laptop it happens often because I only use it during development.

**Short version**: regenerate auth code and try again :sunglasses:

**Comment by ristridin01** [comment]
ok, will keep that monitored...my homebrige is running within a docker container and always online, so it should be stable now.

**Comment by hpruszyn** [comment]
And the problem with duplicated uuid, is it working now? Do you see temperature accessory and is it showing correct value?

**Comment by ristridin01** [comment]
haven´t seen the duplicate message again, but there is still no value shown.
Homekit shows the object but states "not supported", altough manufacture, serial number, modell and firmware are correctly shown.
I have a F2120-12, if that makes the difference with your setup?

**Comment by hpruszyn** [comment]
Yes, that makes a huge difference 😃 

This week I'll release new version with fan and water heating sensors for F1145 and I'll also add configuration to enable debug. In debug mode I'll list parameters retrieved from API and then we make dedicated setup for your nibe model. This is important information because I will create some kind of mapping between parameters from api and heat pomp model to make it easier to add new configs later.

**Comment by ristridin01** [comment]
sounds awesome, I´ll stick with  you!

**Comment by hpruszyn** [comment]
@ristridin01 please update to v0.0.8. You will see error log with instructions - please follow them. Restart is needed after update.

I also think that we can close this one.

**Comment by hpruszyn** [comment]
Bug should not appear any more.
